### PR TITLE
Update Travis: Infrastructure, WordPress core, PHP

### PR DESCRIPTION
**Pull request by dlh01**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,12 +10,23 @@ php:
...
+    - WP_VERSION=4.1 WP_MULTISITE=1
+
+matrix:
+    include:
+        - php: 5.6
+          env: WP_VERSION=master
+        - php: 7.0
+          env: WP_VERSION=master
+    allow_failures:
+        - php: 5.6
```
Comment (mboynes) [inline .travis.yml]: Let's not allow 5.6 to fail

File: .travis.yml [hunk 2]
```diff
@@ -8,12 +10,23 @@ php:
...
-    - WP_VERSION=3.9 WP_MULTISITE=0
-    - WP_VERSION=3.9 WP_MULTISITE=1
-    - WP_VERSION=3.8 WP_MULTISITE=0
-    - WP_VERSION=3.8 WP_MULTISITE=1
+    - WP_VERSION=4.3 WP_MULTISITE=0
+    - WP_VERSION=4.3 WP_MULTISITE=1
+    - WP_VERSION=4.2 WP_MULTISITE=0
+    - WP_VERSION=4.2 WP_MULTISITE=1
+    - WP_VERSION=4.1 WP_MULTISITE=0
+    - WP_VERSION=4.1 WP_MULTISITE=1
```
Comment (mboynes) [inline .travis.yml]: Do you think it's even worth doing 4.1? I think I'd rather remove that, restrict it to 3 versions, and add php 5.6 to the full php version array

**Comment by mboynes** [comment]
Bounced some ideas, but this is good to go regardless :+1:

File: .travis.yml (hunk 2 above)
Comment (dlh01) [inline .travis.yml]: I don't have an opinion about 4.1. I was only following the previous pattern, and I'm not sure what the original reasoning was for that. I'll just drop it and move 5.6.

**Pull request merged**